app/addons/documents/revisionTree/component.react.jsx

```diff
@@ -0,0 +1,332 @@
+      );
+    }
+  });
+
+  var draw = function (paths, deleted, winner, minUniq) {
+    var maxX = grid;
+    var maxY = grid;
+    var levelCount = []; // numer of nodes on some level (pos)
+    lineObjs.length = 0;
+    nodeObjs.length = 0;
+    textObjs.length = 0;
+
+    var map = {}; // map from rev to position
+
+    function drawPath(path) {
+      for (var i = 0; i < path.length; i++) {
+        var rev = path[i];
+        var isLeaf = i === 0;
+        var pos = +rev.split('-')[0];
+
+        if (!levelCount[pos]) {
+          levelCount[pos] = 1;
+        }
+
+        var x = levelCount[pos] * grid;
+        var y = pos * grid;
+
+        if (!isLeaf) {
+          var nextRev = path[i - 1];
+          var nextX = map[nextRev][0];
+          var nextY = map[nextRev][1];
+
+          if (map[rev]) {
+            x = map[rev][0];
+            y = map[rev][1];
+          }
+
+          var lineObj = {
+            "x" : x,
+            "y" : y,
+            "nextX" : nextX,
+            "nextY" : nextY
+          };
+
+          lineObjs.push(lineObj);
+        }
+
+        if (map[rev]) {
+          break;
+        }
+
+        maxX = Math.max(x, maxX);
+        maxY = Math.max(y, maxY);
+        levelCount[pos]++;
+
+        node(x, y, rev, isLeaf, rev in deleted, rev === winner, minUniq);
+        map[rev] = [x, y];
+      }
+    }
+
+    paths.forEach(drawPath);
+  };
+
+  var minUniqueLength = function (arr) {
+    function strCommon(a, b) {
+
+      if (a === b) {
+        return a.length;
+      }
+
+      var i = 0;
+
+      while (++i) {
+        if (a[i - 1] !== b[i - 1]) return i;
+      }
+    }
+
+    var array = arr.slice(0);
+    var com = 1;
+    array.sort();
+
+    for (var i = 1; i < array.length; i++) {
+      com = Math.max(com, strCommon(array[i], array[i - 1]));
+    }
+
+    return com;
+  };
+
+  function node(x, y, rev, isLeaf, isDeleted, isWinner, shortDescLen) {
+    circ(x, y, r, isLeaf, isDeleted, isWinner);
+    var pos = rev.split('-')[0];
+    var id = rev.split('-')[1];
+    var opened = false;
+
+    var textObj = {
+      "stLeft": (x - 40) + "px",
+      "stTop": (y - 30) + "px",
+      "short": pos + '-' + id.substr(0, shortDescLen),
+      "long": pos + '-' + id
+    };
+
+    textObjs.push(textObj);
+  }
+
+  var circ = function (x, y, r, isLeaf, isDeleted, isWinner) {
+
+    var leafStat = "";
+
+    if (isLeaf) {
+      leafStat = "leaf";
+    }
+    if (isWinner) {
+      leafStat = "winner";
+    }
+    if (isDeleted) {
+      leafStat = "deleted";
+    }
+
+    var nodeObj = {
+      "x" : x,
+      "y" : y,
+      "class" : leafStat
+    };
+
+    nodeObjs.push(nodeObj);
+  };
+
+  var App = React.createClass({
+
+    getInitialState: function () {
+      return {
+        lines: [],
+        treeNodes: [],
+        nodeTextObjs: []
+      };
+    },
+
+    componentDidMount: function () {
+      var paths = [];
+      var deleted = {};
+      var treeDataOptions = store.getTreeOptions();
```

Comment:
with flux you should get them using `getInitialState ` - example: https://github.com/apache/couchdb-fauxton/blob/master/app/addons/databases/components.react.jsx#L29-L38

you can then access it in render using `this.state` - example: https://github.com/apache/couchdb-fauxton/blob/master/app/addons/databases/components.react.jsx#L53

the construction of the elements should also happen in render